src/migrateDiagram.js

```diff
@@ -29,16 +29,25 @@ const DMN13URI = '"https://www.omg.org/spec/DMN/20191111/MODEL/"';
  * @returns {Promise<string>}
  */
 export function migrateDiagram(xml) {
+  return new Promise((resolve, reject) => {
+    if (!isString(xml)) {
+      reject(new Error(`expected DMN diagram, found ${ xml }`));
+    }
 
-  if (hasNamespace(DMN12URI, xml)) {
-    return migrateFrom12To13(xml);
-  } else if (hasNamespace(DMN11URI, xml)) {
-    return migrateFrom11To13(xml);
-  } else if (hasNamespace(DMN13URI, xml)) {
-    return Promise.resolve(xml);
-  }
+    if (hasNamespace(DMN11URI, xml)) {
+      try {
+        resolve(migrateFrom11To13(xml));
+      } catch (err) {
+        reject(err);
+      }
+    } else if (hasNamespace(DMN12URI, xml)) {
+      resolve(migrateFrom12To13(xml));
+    } else if (hasNamespace(DMN13URI, xml)) {
+      resolve(xml);
+    }
 
-  throw new Error('unknown namespace');
+    reject(new Error(`expected DMN diagram, found ${ xml }`));
+  });

```

Comment:
BTW this whole function could be async

```suggestion
  if (!isString(xml)) {
    throw new Error(`expected DMN diagram, found ${ xml }`);
  }

  if (hasNamespace(DMN11URI, xml)) {
    return await migrateFrom11To13(xml);
  } else if (hasNamespace(DMN12URI, xml)) {
    return await migrateFrom12To13(xml);
  } else if (hasNamespace(DMN13URI, xml)) {
    return xml;
  }

  throw new Error(`expected DMN diagram, found ${ xml }`);
```

Reply:
This suggestion thing seems to be broken :D

Reply:
As discussed with @nikku we're not going to use `async`. I think the reasoning was that the Camunda products using this module need to support older browsers. Alternatively we could build distros for these browsers.

Reply:
All right, I get it.